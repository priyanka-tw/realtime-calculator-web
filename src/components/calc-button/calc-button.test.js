import React from 'react';
import {act, fireEvent, render} from '@testing-library/react';
import 'jest-styled-components';
import CalcButton from "./calc-button";

describe('Calc Button', () => {
    const mockFn = jest.fn();

    it('should render the component with given value and default props', () => {
        const {container} = render(<CalcButton value="1" onClick={mockFn}/>);

        expect(container).toMatchSnapshot();
    });

    it('should render the component with given value as an operation', () => {
        const {container} = render(<CalcButton value="+" isOperation/>);

        expect(container).toMatchSnapshot();
    });


    it('should render the component for equal button', () => {
        const {container} = render(<CalcButton value="=" isEqualTo/>);

        expect(container).toMatchSnapshot();
    });

    it('should perform the given action on click of button', async () => {
        const {getByTestId} = render(<CalcButton value="1" onClick={mockFn}/>);

        await act(async () => {
            await fireEvent.click(getByTestId('digit-button-1'));
        });

        expect(mockFn).toHaveBeenCalledTimes(1);
    });

    it('should render the component for zero button', async () => {
        const {container} = render(<CalcButton value="0" isZero/>);

        expect(container).toMatchSnapshot();
    });
});
