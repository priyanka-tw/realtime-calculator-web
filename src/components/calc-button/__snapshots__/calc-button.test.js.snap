// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calc Button should render the component for equal button 1`] = `
.c0 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  background-color: #ffc107;
}

<div>
  <button
    class="c0"
    data-testid="digit-button-="
    name="="
    type="button"
  >
    =
  </button>
</div>
`;

exports[`Calc Button should render the component for zero button 1`] = `
.c0 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  min-width: 120px;
}

<div>
  <button
    class="c0"
    data-testid="digit-button-0"
    name="0"
    type="button"
  >
    0
  </button>
</div>
`;

exports[`Calc Button should render the component with given value and default props 1`] = `
.c0 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
}

<div>
  <button
    class="c0"
    data-testid="digit-button-1"
    name="1"
    type="button"
  >
    1
  </button>
</div>
`;

exports[`Calc Button should render the component with given value as an operation 1`] = `
.c0 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  color: #ffc107;
}

<div>
  <button
    class="c0"
    data-testid="digit-button-+"
    name="+"
    type="button"
  >
    +
  </button>
</div>
`;
